### PR TITLE
ci: implement ghcr.io expiration for images and cache layers

### LINKED ISSUE #568: Implement ghcr.io expiration for images and cache layers
It appears that ghcr.io does not support setting expiration. The only way is to periodically delete old images. This is probably a good thing to do, just to avoid excessive storage usage.

Although there does not seem to be any limit for us, it seems to be a good practice to clean up after ourselves.

* https://github.com/WoozyMasta/quay-expiration
* https://github.com/eumel8/expire-action
* https://github.com/snok/container-retention-policy/blob/main/main.py

### PR DESCRIPTION
**Pull request by jiridanek**
## Description

Fixes #568

## How Has This Been Tested?

See the action in action at https://github.com/jiridanek/notebooks/actions/runs/9717613825/job/26823642202#step:3:18

The repos this is cleaning are https://github.com/jiridanek/notebooks/pkgs/container/notebooks%2Fworkbench-images and https://github.com/jiridanek/notebooks/pkgs/container/notebooks%2Fworkbench-images%2Fbuild-cache

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by caponetto — APPROVED** [review]

File: .github/workflows/purge-ghcr.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+permissions:
+  packages: write
+
+jobs:
+  clean:
+    runs-on: ubuntu-latest
+    name: Delete old test images
+    steps:
+      # https://github.com/snok/container-retention-policy?tab=readme-ov-file#parameters
+      - uses: snok/container-retention-policy@v3.0.0
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply .github/workflows/purge-ghcr.yaml:23]: yeah, why not! thanks

**Review by caponetto — APPROVED** [review]

**Comment by atheo89** [comment]
Are we sure that sufficient time has been allocated to inspect the Trivy report before this workflow proceed to prune the images?

**Comment by caponetto** [comment]
> Are we sure that sufficient time has been allocated to inspect the Trivy report before this workflow proceed to prune the images?

I don't see it as an issue since the workflow prunes images that are 3 weeks old. We can always resort to the latest report/image if necessary.

**Review by jstourac — APPROVED** [review]

**Comment by jiridanek** [comment]
we could push `:latest` tags to ghcr and set the workflow to leave the newest image around, if we wanted to spend more time on this

**Comment by atheo89** [comment]
>the workflow prunes images that are 3 weeks old

Ahhh ok i got it! 

/lgtm
/approve

**Pull request merged**